test/unit/collector/remote-method.test.js

```diff
@@ -261,6 +262,16 @@ tap.test('when the connection fails', (t) => {
   })
 
   t.test('should correctly handle a DNS lookup failure', (t) => {
+    const lookup = dns.lookup
```

Comment:
this is fine but could just use sinon too

```js
    const err = Error('no dns')
    err.code = dns.NOTFOUND
    sinon.stub(dns, 'lookup').yields(err)
    t.teardown(() => {
      dns.lookup.restore()
    })
```

Reply:
not really. we typically use sinon for mocking but older tests will still have patching style like this. I just wanted to call it out, feel free to merge

Reply:
I am not at all familiar with the sinon API. I like the least amount of mystery in tests as possible.

Reply:
that's fair. I suggest you familiarize yourself as this repo does use it heavily. but again not a big deal here